card/mod/core/set/all/name_events.rb

```diff
@@ -14,6 +14,7 @@
 event :validate_name, :validate, on: :save, changed: :name do
   validate_legality_of_name
   return if errors.any?
+  Card.write_to_soft_cache self
```

Comment:
Make sure we have the current card in the cache (this wasn't happening in Card.create).

I do it here, because I think it only really helps if we do it when we have a valid name.  But there's probably a smarter way...